parsl/curvezmq.py

```diff
@@ -0,0 +1,193 @@
+import os
+from abc import ABCMeta, abstractmethod
+from typing import Optional, Tuple, Union
+
+import zmq
+import zmq.auth
+from zmq.auth.thread import ThreadAuthenticator
+
+
+def ensure_certificates(base_dir: Union[str, os.PathLike]):
+    """Create server and client certificates in a private directory.
+    If the directory already exists, ensure that only the owner has
+    read/write/execute permissions then return immediately.
+
+    Parameters
+    ----------
+    base_dir : str | os.PathLike
+        Parent directory of the private certificates directory.
+    """
+    cert_dir = os.path.join(base_dir, "certificates")
+    try:
+        os.mkdir(cert_dir)
+    except FileExistsError:
+        return cert_dir
+    finally:
+        os.chmod(cert_dir, 0o700)
```

Comment:
We want to return before hitting `zmq.auth.create_certificates` since this will overwrite the existing certs. When eventually implemented, multiple executors could call this function per run. 

I'll add a comment to clear this up a bit.

Reply:
Following some out-of-band discussion, we've decided to generate separate certificates for each executor and, eventually, separate certificates for the monitoring hub channels, which we will likely pass to the executors from the DFK. This will simplify the certificate creation logic discussed above.